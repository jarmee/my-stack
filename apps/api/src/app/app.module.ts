import { Module } from '@nestjs/common';
import { ConfigModule, ConfigService } from '@nestjs/config';
import { TypeOrmModule } from '@nestjs/typeorm';

import { BuildInfoModule } from './api/build-info';
import { Stack, StacksModule } from './api/stacks';
import { StackTechnology } from './api/stacks/technologies';

@Module({
  imports: [
    ConfigModule.forRoot({
      envFilePath: ['.env.development.local', '.env.development'],
    }),
    TypeOrmModule.forRootAsync({
      imports: [ConfigModule],
      useFactory: (config: ConfigService) => ({
        type: 'postgres',
        host: config.get('DB_HOST') ?? 'localhost',
        port: config.get('DB_PORT') ?? 5432,
        username: config.get('DB_USER') ?? 'root',
        password: config.get('DB_PASS') ?? 'secret',
        database: 'mystack',
        entities: [Stack, StackTechnology],
        synchronize: true,
      }),
      inject: [ConfigService],
    }),
    BuildInfoModule,
    StacksModule,
  ],
})
export class AppModule {}
