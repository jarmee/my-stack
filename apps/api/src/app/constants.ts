export const API = {
  ENDPOINTS: {
    STACKS: 'stacks',
    STACKS_TECHNOLOGIES: (stackId: number | string = ':stackId') =>
      `stacks/${stackId}/technologies`,
    BUILD_INFO: 'build-info',
  },
};
