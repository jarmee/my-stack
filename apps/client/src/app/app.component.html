<mat-toolbar color="primary">
  <button
    mat-icon-button
    data-test-id="navigtationButton"
    aria-label="Open Navigation"
    (click)="onToggleNavigation()"
  >
    <mat-icon>menu</mat-icon>
  </button>
  <span class="mr-4">{{ title }}</span>
</mat-toolbar>
<main>
  <mat-drawer-container>
    <mat-drawer
      data-test-id="sidebar"
      mode="side"
      [opened]="isSidebarNavigationOpen"
    ></mat-drawer>
    <mat-drawer-content>
      <router-outlet></router-outlet>
    </mat-drawer-content>
  </mat-drawer-container>
</main>
