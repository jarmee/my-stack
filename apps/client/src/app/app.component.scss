:host {
  display: block;
  height: 100vh;
  width: 100vw;

  mat-drawer-container {
    height: 100vh;

    mat-drawer {
      width: 248px;
    }
  }
}
