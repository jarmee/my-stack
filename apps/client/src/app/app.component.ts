import { Component } from "@angular/core";
import { MatButtonModule } from "@angular/material/button";
import { MatIconModule } from "@angular/material/icon";
import { MatSidenavModule } from "@angular/material/sidenav";
import { MatToolbarModule } from "@angular/material/toolbar";
import { RouterModule } from "@angular/router";

@Component({
  standalone: true,
  imports: [
    MatToolbarModule,
    MatSidenavModule,
    MatButtonModule,
    MatIconModule,
    RouterModule,
  ],
  selector: "mystack-root",
  templateUrl: "./app.component.html",
  styleUrls: ["./app.component.scss"],
})
export class AppComponent {
  title = "My Stack";

  toggleSidebarNavigation = false;

  get isSidebarNavigationOpen(): boolean {
    return this.toggleSidebarNavigation;
  }

  onToggleNavigation() {
    this.toggleSidebarNavigation = !this.toggleSidebarNavigation;
  }
}
