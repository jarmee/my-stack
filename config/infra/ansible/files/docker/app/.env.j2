MY_STACK_DEPLOYMENT={{ deployment.key }}
MY_STACK_VERSION={{ deployment.value.version }}
MY_STACK_PREFIX={{ deployment.value.prefix }}

DB_PORT=5432
DB_USER=root
DB_PASS={{ vault_db_password }}
