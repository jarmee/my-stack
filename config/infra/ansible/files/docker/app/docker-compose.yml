version: '3'
services:
  my-stack-db:
    image: ghcr.io/jarmee/my-stack-db:${MY_STACK_VERSION}
    container_name: my-stack-db-${MY_STACK_DEPLOYMENT}
    restart: on-failure:5
    networks:
      - traefik-proxy
    environment:
      POSTGRES_DB: 'mystack'
      POSTGRES_USER: '${DB_USER}'
      POSTGRES_PASSWORD: '${DB_PASS}'
    volumes:
      - ./db-data:/data/db
    labels:
      - 'traefik.enable=false'

  my-stack-client:
    image: ghcr.io/jarmee/my-stack-client:${MY_STACK_VERSION}
    container_name: my-stack-client-${MY_STACK_DEPLOYMENT}
    restart: on-failure:5
    networks:
      - traefik-proxy
    labels:
      - 'traefik.http.routers.my-stack-client-${MY_STACK_DEPLOYMENT}.rule=Host(`${MY_STACK_PREFIX}mystack.letz.dev`)'

  my-stack-api:
    image: ghcr.io/jarmee/my-stack-api:${MY_STACK_VERSION}
    container_name: my-stack-api-${MY_STACK_DEPLOYMENT}
    restart: on-failure:5
    networks:
      - traefik-proxy
    environment:
      DB_HOST: 'my-stack-db'
      DB_PORT: '${DB_PORT}'
      DB_USER: '${DB_USER}'
      DB_PASS: '${DB_PASS}'
    labels:
      - 'traefik.http.routers.my-stack-api-${MY_STACK_DEPLOYMENT}.rule=Host(`${MY_STACK_PREFIX}mystack.letz.dev`) && PathPrefix(`/api`)'

networks:
  traefik-proxy:
    external: true
