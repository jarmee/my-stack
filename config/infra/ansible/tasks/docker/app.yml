- name: creates directory for docker-compose files
  file:
    path: "{{ ansible_env.HOME }}/app/{{ deployment.key }}"
    state: directory
    owner: "{{ ansible_user_id }}"
    group: "{{ group }}"
    mode: 0700

- name: copies docker-compose.yml to stack folder
  copy:
    src: "files/docker/app/docker-compose.yml"
    dest: "{{ ansible_env.HOME }}/app/{{ deployment.key }}/docker-compose.yml"
    owner: "{{ ansible_user_id }}"
    group: "{{ group }}"
    mode: 0600
  register: compose

- name: templates .env to stack folder
  template:
    src: "files/docker/app/.env.j2"
    dest: "{{ ansible_env.HOME }}/app/{{ deployment.key }}/.env"
    owner: "{{ ansible_user_id }}"
    group: "{{ group }}"
    mode: 0600
  register: env

- name: start containers
  command: "docker compose up -d --remove-orphans"
  args:
    chdir: "{{ ansible_env.HOME }}/app/{{ deployment.key }}"
  when: env.changed or compose.changed
