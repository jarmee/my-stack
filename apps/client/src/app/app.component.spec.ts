import { ComponentFixture, TestBed } from "@angular/core/testing";
import { By } from "@angular/platform-browser";
import { NoopAnimationsModule } from "@angular/platform-browser/animations";

import { AppComponent } from "./app.component";

const CssSelectors = {
  ToggleNavigationButton: By.css('[data-test-id="navigtationButton"]'),
  Sidebar: By.css('[data-test-id="sidebar"]'),
};

describe("AppComponent", () => {
  let fixture: ComponentFixture<AppComponent>;
  let component: AppComponent;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent, NoopAnimationsModule],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
  });

  it("should create", () => {
    expect(component).toBeTruthy();
  });

  it("should match snapshot", () => {
    expect(fixture).toMatchSnapshot();
  });

  describe("onToggleNavigation", () => {
    it("should open the navigation", () => {
      component.toggleSidebarNavigation = false;

      fixture.debugElement
        .query(CssSelectors.ToggleNavigationButton)
        .triggerEventHandler("click", {});
      fixture.detectChanges();

      expect(
        fixture.debugElement.query(CssSelectors.Sidebar).componentInstance
          .opened
      ).toBe(true);
    });

    it("should close the navigation", () => {
      component.toggleSidebarNavigation = true;

      fixture.debugElement
        .query(CssSelectors.ToggleNavigationButton)
        .triggerEventHandler("click", {});
      fixture.detectChanges();

      expect(
        fixture.debugElement.query(CssSelectors.Sidebar).componentInstance
          .opened
      ).toBe(false);
    });
  });
});
