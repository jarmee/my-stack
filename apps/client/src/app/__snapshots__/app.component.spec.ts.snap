// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<mystack-root
  __ngContext__={[Function Number]}
  title={[Function String]}
  toggleSidebarNavigation="false"
>
  <mat-toolbar
    class="mat-toolbar mat-primary"
    color="primary"
  >
    <button
      aria-label="Open Navigation"
      class="mdc-icon-button mat-mdc-icon-button"
      data-test-id="navigtationButton"
      mat-icon-button=""
    >
      <span
        class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
      />
      <mat-icon
        aria-hidden="true"
        class="mat-icon notranslate"
        role="img"
      >
        menu
      </mat-icon>
      <span
        class="mat-mdc-focus-indicator"
      />
      <span
        class="mat-ripple mat-mdc-button-ripple"
        matripple=""
      />
      <span
        class="mat-mdc-button-touch-target"
      />
    </button>
    <span
      class="mr-4"
    >
    </span>
  </mat-toolbar><main>
    <mat-drawer-container
      class="mat-drawer-container"
    >
      <mat-drawer
        class="mat-drawer ng-trigger ng-trigger-transform"
        data-test-id="sidebar"
        mode="side"
        tabindex="-1"
      >
        <div
          cdkscrollable=""
          class="mat-drawer-inner-container"
        />
      </mat-drawer>
      <mat-drawer-content
        class="mat-drawer-content"
      >
        <router-outlet />
      </mat-drawer-content>
    </mat-drawer-container>
  </main>
</mystack-root>
`;
